src/app/track/track.component.ts

```diff
@@ -1,346 +1,476 @@
-      false, false, false, false, 0, null);
-    a.dispatchEvent(e);
-    a.remove();
-    this.isExporting = false;
-  }
-  exportCSV() {
-    this.isExporting = true;
-    let fileContents = 'Date;Time;Latitude;Longitude;Course;Speed;Right;Left;Confirmed;insidePrivacyArea\n';
-    for (const p of this.trackData.points) {
-      fileContents += [p.date, p.time, p.latitude, p.longitude, p.course, p.speed, p.d1, p.d2, p.flag, p.private].map(
-        (value) => {
-          if (value == null) {
-            return '';
-          } else if (typeof value === 'number') {
-            return String(value);
-          } else if (typeof value === 'boolean') {
-            return value ? '1' : '0';
-          } else if (typeof value === 'string' && /[^a-zA-Z0-9_.,: -]/.test(value)) {
-            // properly escape strings that contain quotes, semicolons, or
-            // anything else that is weird
-            return JSON.stringify(value);
-          } else {
-            // unproblematic string can be returned as-is
-            return value;
-          }
+
+        const layerSwitcher = new LayerSwitcher({
+            groupSelectStyle: 'children',
+            startActive: true,
+            activationMode: 'click',
+            reverse: false,
+
+
+        });
+        olMap.addControl(layerSwitcher);
+
+
+        function pointStyleFunction(feature, resolution) {
+            let distance = feature.get("distance");
+            let radius = 200 / resolution;
+
+            return new Style({
+                image: new CircleStyle({
+                    radius: radius < 20 ? radius : 20,
+                    fill: evaluateDistanceForFillColor(distance),
+                    stroke: evaluateDistanceForStrokeColor(distance)
+                }),
+                text: createTextStyle(distance, resolution),
+            });
+        }
+
+
+        const evaluateDistanceForFillColor = function (distance) {
+            let redFill = new Fill({color: 'rgba(255, 0, 0, 0.2)'})
+            let orangeFill = new Fill({color: 'rgba(245,134,0,0.2)'})
+            let greenFill = new Fill({color: 'rgba(50, 205, 50, 0.2)'})
+
+            switch (evaluateDistanceColor(distance)) {
+                case 'red': return redFill
+                case 'orange': return orangeFill
+                case 'green': return greenFill
+            }
+        }
+
+        const evaluateDistanceForStrokeColor = function (distance) {
+            let redStroke = new Stroke({color: 'rgb(255, 0, 0)'})
+            let orangeStroke = new Stroke({color: 'rgb(245,134,0)'})
+            let greenStroke = new Stroke({color: 'rgb(50, 205, 50)'})
+
+            switch (evaluateDistanceColor(distance)) {
+                case 'red': return redStroke
+                case 'orange': return orangeStroke
+                case 'green': return greenStroke
+            }
+        }
+
+        const evaluateDistanceColor = function (distance) {
+            const distanceLimit1 = 200
```

Comment:
We should think about creating a file with all the constants in it to avoid having "configuration" of behaviour spread over many files